### ISSUE TITLE
Proposal to add `metadata` section to Missions, Runtimes and Versions

### ISSUE THREAD

**Issue by quintesse** [fabric8-launcher/launcher-booster-catalog-service#30]
I'm seeing several things in the current UI design and the new one that are not properly represented in the `metadata.yaml` file. One example is the "tech preview" ribbon. In the new UI I see several different kinds of small icons that appear next to missions or runtimes.

Hard-coding these into the format will make the format very inflexible IMO. So I'm suggesting we add `metadata` sections to all of them just like we did with Boosters. The `metadata` section would just by copied as-is right into the result of the Rest API. That way runtime teams / booster authors can add whatever they like and in the UI they can also show whatever they want depending on that info.

Wdyt @gastaldi @edewit @ia3andy ?

**Comment by gastaldi** [comment]
+1, that looks like a good idea

**Comment by gastaldi** [comment]
In the issue you mention `metadata.yaml`, but we haven't migrated from the JSON format, I assume that's something that needs to be done first, right?

**Comment by quintesse** [comment]
That's already done and in master
